packages/analytics/analytics-utilities/src/queryTime.ts

```diff
@@ -79,11 +81,16 @@ abstract class BaseQueryTime implements QueryTime {
 export class TimeseriesQueryTime extends BaseQueryTime {
   private readonly granularity: GranularityValues
 
-  constructor(timeframe: Timeframe, granularity?: GranularityValues, tz?: string) {
-    super(timeframe, tz)
+  constructor(timeframe: Timeframe, granularity?: GranularityValues, tz?: string, dataGranularity?: GranularityValues, fineGrain?: boolean) {
+    super(timeframe, tz, dataGranularity)
 
-    if (granularity && timeframe.allowedGranularities().has(granularity)) {
+    if (granularity && timeframe.allowedGranularities(fineGrain).has(granularity)) {
       this.granularity = granularity
+    } else if (fineGrain) {
+      // TODO: when removing the feature flag, consider redefining `defaultResponseGranularity`
+      // in the timeframes constructor: it should probably handle this calculation on its own.
+      const finestGranularity = timeframe.allowedGranularities(fineGrain).keys().next().value
```

Comment:
nit: feels almost like this should just be a list instead of a Set

Reply:
Hmm.  I think that's fair, but would prefer not to change it right now since it's a breaking change to the API.  I think a lot of this needs an overhaul, and plan to advocate for that overhaul when we do fine-grained absolute time.